setup.py

```diff
@@ -1,167 +1,63 @@
+version = get_version(os.path.join(name, '_version.py'))
 
-class NPM(Command):
-    description = 'install package.json dependencies using npm'
+js_dir = os.path.join(here, 'js')
 
-    user_options = []
+# Representative files that should exist after a successful build
+jstargets = [
+    os.path.join(js_dir, 'dist', 'index.js'),
+]
 
-    node_modules = pjoin(node_root, 'node_modules')
+data_files_spec = [
+    ('share/jupyter/nbextensions/jupyter-matplotlib',
+     'ipympl/nbextension', '*.*'),
+    ('share/jupyter/labextensions/jupyter-matplotlib',
+     'ipympl/labextension', "*.*"),
+    ('etc/jupyter/nbconfig/notebook.d', '.', 'jupyter-matplotlib.json'),
+]
 
-    targets = [
-        pjoin(here, 'ipympl', 'static', 'extension.js'),
-        pjoin(here, 'ipympl', 'static', 'index.js'),
-        pjoin(here, 'ipympl', 'static', 'package.json')
-    ]
-
-    def initialize_options(self):
-        pass
-
-    def finalize_options(self):
-        pass
-
-    def has_npm(self):
-        try:
-            check_call(['npm', '--version'])
-            return True
-        except Exception:
-            return False
-
-    def run(self):
-        has_npm = self.has_npm()
-        if not has_npm:
-            log.error("`npm` unavailable.  If you're running this command "
-                      "using sudo, make sure `npm` is available to sudo")
-
-        env = os.environ.copy()
-        env['PATH'] = npm_path
-
-        if self.has_npm():
-            log.info("Installing build dependencies with npm.  "
-                     "This may take a while...")
-            check_call(['npm', 'install'], cwd=node_root, stdout=sys.stdout,
-                       stderr=sys.stderr)
-            check_call(['npm', 'pack'], cwd=node_root, stdout=sys.stdout,
-                       stderr=sys.stderr)
-
-        for t in self.targets:
-            if not os.path.exists(t):
-                msg = 'Missing file: %s' % t
-                if not has_npm:
-                    msg += ('\nnpm is required to build a development version '
-                            'of widgetsnbextension')
-                raise ValueError(msg)
-
-        # update package data in case this created new files
-        update_package_data(self.distribution)
-
-
-version_ns = {}
-with open(pjoin(here, 'ipympl', '_version.py')) as f:
-    exec(f.read(), {}, version_ns)
+cmdclass = create_cmdclass('jsdeps', data_files_spec=data_files_spec)
+cmdclass['jsdeps'] = combine_commands(
+    install_npm(js_dir, build_cmd='build'), ensure_targets(jstargets),
+)
 
 setup_args = dict(
     name=name,
-    version=version_ns['__version__'],
+    version=version,
     description='Matplotlib Jupyter Extension',
-    long_description=LONG_DESCRIPTION,
+    long_description=long_description,
     license='BSD License',
     include_package_data=True,
-    data_files=get_data_files(),
     install_requires=[
         'ipykernel>=4.7',
         'ipywidgets>=7.5.0',
         'matplotlib>=2.0.0'
     ],
```

Comment:
Now that https://github.com/jupyter-widgets/ipywidgets/pull/2995 was merged we can do this!!:
```suggestion
    install_requires=[
        'ipykernel>=4.7',
        'ipywidgets>=7.5.0',
        'matplotlib>=2.0.0',
        'jupyterlab_widgets>=1.0.0a6'
    ],
```